pyecoregen/templates/module.py.tpl

```diff
@@ -20,12 +20,49 @@ eClassifiers = {}
 getEClassifier = partial(Ecore.getEClassifier, searchspace=eClassifiers)
 
 {%- for c in element.eClassifiers if c is type(ecore.EEnum) -%}
+{% if loop.first and textX %}
+class EEnum(Ecore.EEnum):
+
+    def __init__(self, name=None, default_value=None, literals=None, **kwargs):
+        super().__init__(name, default_value, literals)
+        self.__dict__.update(kwargs)
+
+    @property
+    def value(self):
+        ''' returns the value of the attribute '''
+        for attr in [a for a in self.__dict__ if
+                     not a.startswith('__') and not
+                     a.startswith('_tx_')]:
+            obj = getattr(self, attr)
+            if obj and isinstance(obj, str):
+                return obj
+{% endif %}
 {{ modutil.generate_enum(c) }}
 {%- endfor %}
 {% for c in element.eClassifiers if c is type(ecore.EDataType) -%}
+{% if loop.first and textX %}
```

Comment:
Wouldn't it be easier to simply generates a `MyDataTypeName.__name__ = 'MyDataTypeName'` for each `EDataType` generation?

Reply:
Yes, would be way easier. I will change that.